internal/bminventory/inventory.go

```diff
@@ -4263,13 +4263,21 @@ func (b *bareMetalInventory) refreshHostAndClusterStatuses(
 
 	var err error
 
+	infraEnv, err := common.GetInfraEnvFromDB(b.db, *clusterID)
```

Comment:
done

Reply:
I don't think we should re-fetch it from the DB.
Pls have a look at the caller of `refreshHostAndClusterStatuses` and see that in both cases the host was already retrieved from the DB, therefore I'd suggest to change the signature of this function to accept the entire host struct and use the value of the infraenv-id from it.